plugins/nanospell/plugin.js

```diff
@@ -853,28 +848,25 @@
 			}
 
 			function startCheckOrMarkWords(words, blockList) {
+				var selectionStart = editor.getSelection().getStartElement();
 				if (words.length > 0) {
 					editor.fire(EVENT_NAMES.START_CHECK_WORDS, {
 						words: words,
 						blockList: blockList
 					});
 				}
 				else {
-					editor.lockSelection();
-					var bookmarks = editor.getSelection().createBookmarks(true);
 					for (var i = 0; i < blockList.length; i++) {
 						var rootElement = blockList[i];
 
 						editor.fire(
 							EVENT_NAMES.START_RENDER,
 							{
 								root: rootElement,
-								needsBookmarkCreated: false,
+								needsBookmarkCreated: selectionStart ? rootElement.contains(selectionStart) || rootElement.equals(selectionStart) : null,
```

Comment:
Same story, here. We can do the dom-walking one time outside this loop and make the inner-loop logic more performant.